component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisIterator.java

```diff
@@ -0,0 +1,286 @@
+            resultMap.clear();
+        }
+        result = null;
+        scanResultsList = null;
+        scanResults = null;
+    }
+
+    @Override
+    public boolean hasNext() {
+        if (!this.preFetched) {
+            // check whether object is pre fetched
+            this.nextValue = this.next();
+            this.preFetched = true;
+        }
+        return nextValue != null;
+    }
+
+    @Override
+    public Object[] next() {
+        if (this.preFetched) {
+            this.preFetched = false;
+            Object[] record = this.nextValue;
+            this.nextValue = null;
+            return record;
+        }
+        List<Object[]> finalResult;
+        if (initialTraverse) {
+            finalResult = fetchResults();
+            initialTraverse = false;
+            this.iterator = finalResult.listIterator();
+            if (finalResult.isEmpty()) {
+                this.closeImpl();
+                return null;
+            }
+        }
+        if (!iterator.hasNext()) {
+            scanResultsList.clear();
+            finalResult = fetchResults();
+            this.iterator = finalResult.listIterator();
+            if (finalResult.isEmpty()) {
+                this.closeImpl();
+                return null;
+            }
+            return iterator.next();
+        } else {
+            return iterator.next();
+        }
+
+    }
+
+    @Override
+    public void remove() {
+        //Do nothing. This is a read only iterator
+    }
+
+    private List<Object[]> fetchResults() {
+        List<Object[]> resultList = new ArrayList<>();
+
+        if (Objects.nonNull(query) &&
+                !("true".equalsIgnoreCase((query.getStreamVariable()).getName()))) {
+            return fetchResultsWithCondition();
+
+        } else if (Objects.nonNull(query) &&
+                ("true".equalsIgnoreCase((query.getStreamVariable()).getName()))) {
+
+            return fetchAllResults();
+        }
+        return resultList;
+    }
+
+    private List<Object[]> fetchResultsWithCondition() {
+        List<Object[]> resultList = new ArrayList<>();
+
+        storeVariable = query
+                .getStoreVariable();
+        streamVariable = query
+                .getStreamVariable();
+        if (indexes.contains(storeVariable.getName())) {
+            return fetchResultsByIndexedValue();
+        } else if (primaryKeys.contains(storeVariable.getName())) {
+            return fetchResultsByPrimaryKey();
+        }
+        return resultList;
+    }
+
+    private List<Object[]> fetchResultsByPrimaryKey() {
+        List<Object[]> resultList = new ArrayList<>();
+        if (isInitialTraverse) {
+            resultMap = jedis.hgetAll(tableName + ":" + streamVariable.getName());
+            if (Objects.nonNull(resultMap) && !resultMap.isEmpty()) {
+                result = resultsGenerator(resultMap);
+                resultList.add(result);
+                isInitialTraverse = false;
+            }
+        } else {
```

Comment:
I think we can remove this else part. Because we always return the resultList without considering the "isInitialTraverse" status

Reply:
Fixed by eef9f6d